packages/web3-providers-http/src/index.ts

```diff
@@ -161,6 +194,39 @@ export default class Web3ProvidersHttp
                 },
                 args.providerOptions?.axiosConfig || {}
             );
+            // If the above call was successful, then we're connected
+            // to the client, and should emit accordingly (EIP-1193)
+            // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md#connect-1
+            if (this._connected === false) this._connectToClient();
+            return response.data.data ? response.data.data : response.data;
+        } catch (error) {
+            if (error.code === 'ECONNREFUSED' && this._connected) {
+                this._connected = false;
+                // TODO replace with ProviderRpcError
+                this.emit(Web3ProviderEvents.Disconnect, { code: 4900 });
+            }
+            throw Error(error.message);
+        }
+    }
+
+    /**
+     * Makes an Axios POST request using provided {args}
+     *
+     * @param args RPC options, request params, AxiosConfig
+     * @returns
+     */
+    async request(
+        args: Eth1RequestArguments | Eth2RequestArguments
+    ): Promise<RpcResponse | any> {
```

Comment:
We shouldn't have `| any` here

Reply:
We can't just remove `| any` though, because the beaconchain responses don't match the `RpcResponse` that we see in ETH1
![image](https://user-images.githubusercontent.com/20375610/130682555-39e82239-ecc7-4b04-8636-8e05f5b2fea7.png)

Reply:
So I think we need to create a type of `eth2RpcResponse` (we'll rename `eth2` later) that looks like:

```javascript
export type eth2RpcResponse = { [key: string]: any} | { [key: string]: any}[]
```